worksheet.rb

```diff
@@ -22,13 +79,154 @@
+#             rider_id: "RD0066",
+#             date: "5th Feb 2016",
+#             cost: 35,
+#             rating: 3
+#         }
+#     ],
+#     driver_DR0003: [
+#         {
+#             driver_id: "DR0003",
+#             rider_id: "RD0003",
+#             date: "5th Feb 2016",
+#             cost: 50,
+#             rating: 2
+#         },
+#         {
+#             driver_id: "DR0003",
+#             rider_id: "RD0003",
+#             date: "5th Feb 2016",
+#             cost: 50,
+#             rating: 2
+#         }
+#     ],
+#     driver_DR0004: [
+#         {
+#             driver_id: "DR0004",
+#             rider_id: "RD0003",
+#             date: "3rd Feb 2016",
+#             cost: 5,
+#             rating: 5
+#         },
+#         {
+#             driver_id: "DR0004",
+#             rider_id: "RD0022",
+#             date: "3rd Feb 2016",
+#             cost: 10,
+#             rating: 4
+#
+#         },
+#         {
+#             rider_id: "RD0073",
+#             date: "5th Feb 2016",
+#             cost: 20,
+#             rating: 5
+#         }
+#     ]
+# }
 
 ########################################################
 # Step 4: Total Driver's Earnings and Number of Rides
 
 # Use an iteration blocks to print the following answers:
 # - the number of rides each driver has given
+# answer:
+# driver_DR0001, ride number: 3
+# driver_DR0002, ride number: 3
+# driver_DR0003, ride number: 2
+# driver_DR0004, ride number: 3
+# ----------------------
 # - the total amount of money each driver has made
+# answer:
+# driver_DR0001, earned: $85
+# driver_DR0002, earned: $75
+# driver_DR0003, earned: $100
+# driver_DR0004, earned: $35
+# --------------------------
 # - the average rating for each driver
+# answer:
+# driver_DR0001, average rating: 3
+# driver_DR0002, average rating: 3
+# driver_DR0003, average rating: 2
+# driver_DR0004, average rating: 4
+# --------------------------
 # - Which driver made the most money?
-# - Which driver has the highest average rating?
\ No newline at end of file
+# answer: driver_DR0003, made the most money.
+# - Which driver has the highest average rating?
+# answer: driver_DR0004, has the highest average rating.
+#
+# code:
+# #initiated a new hash to store drivers id & total earning
+# total_earning = Hash.new(0)
+# #initiated a new hash to store drivers id & total earning
+# average_rating = Hash.new(0)
+#
+# # for each driver
+# ride_history.each do |driver|
+#   # average rating for each driver
+#   rating_average = driver[1].map { |x| x[:rating]}.sum / driver[1].size
+#
+#   total_cost = 0
+#   # for each ride for that driver
+#   driver[1].each do |i|
+#     total_cost += i[:cost]
+#   end
+#   #key value pairs added to the hash(total_earning)
+#   total_earning[driver[0]] = total_cost
+#   average_rating[driver[0]] = rating_average
+#
+#   #summery:
+#   puts "#{driver[0]}, ride number: #{driver[1].length} earned: $#{total_cost} - average rating: #{rating_average}"
```

Comment:
We were hoping for this to be turned in as code. :P Not a big deal, I got it working on my end.